scripts/prepare_docker_buildinfo.sh

```diff
@@ -55,14 +55,27 @@ ENV IMAGENAME='${IMAGENAME}'
 ENV DISTRO='${DISTRO}'
 RUN pre_run_buildinfo '${IMAGENAME}'
 '
+DOWNLOADED_DOCKER_IMAGES_DOCKERFILE_PRE_SCRIPT='# Auto-Generated for buildinfo
```

Comment:
This file don't need to change. Downloaded docker image will not call this script.

Reply:
@liushilongbuaa - the AMD team can share the file/private URL with you.  How do they do this?

Reply:
I mean prepare_docker_buildinfo.sh is designed to control docker image package versions.
In this scenario, we don't need this feature.
Just change your Dockerfile to oneline(other steps move to docker-dpu-base):
FROM docker-dpu-base
Then at the beginning of prepare_docker_buildinfo.sh, check the $2 line count. If it is 1, then exit.

Reply:
I tried out this change before. I get compilation error with it. Are there any dependencies that cannot be skipped in prepare_docker_buildinfo.sh
```
[Warning] One or more build-args [SONIC_VERSION_CACHE docker_container_name frr_user_uid image_version user SONIC_VERSION_CACHE_SOURCE frr_user_gid guid uid] were not consumed
error squashing image: error getting tar stream to parent: layer ID 'sha256:bb88c4ad01631bb3e6fd8a9ea1721e328bf5fb205f1709663799805a2eff29df' is not a parent of the specified layer: cannot provide diff to non-parent
```

Reply:
@lguohan , 'docker build --squash' squash options required Dockerfile to have at least one RUN command.
I checked docker doc. '--squash' is still an experimental feature not stable feature. Can we remove squash option?

Reply:
@shanshri , you used DOWNLOADED_DOCKER_IMAGES as a tag to run different script.
You can use a tag to 'return' at the beginning of prepare_docker_buildinfo.sh.
SKIP_HOOK is a better name.

Reply:
@liushilongbuaa i am exiting at the beginning of prepare_docker_buildinfo.sh

```
diff --git a/scripts/prepare_docker_buildinfo.sh b/scripts/prepare_docker_buildinfo.sh
index 15d9cfe2f..5411302cd 100755
--- a/scripts/prepare_docker_buildinfo.sh
+++ b/scripts/prepare_docker_buildinfo.sh
@@ -17,6 +17,11 @@ ARCH=$3
 DOCKERFILE_TARGET=$4
 DISTRO=$5

+NUM_OF_LINES=$(wc -l $DOCKERFILE | cut -d ' ' -f1)
+if [[ $NUM_OF_LINES == 1 ]]; then
+       exit 0
+fi
+
```

Reply:
It is because we use '--squash' options in docker build.
'--squash' options need at least one RUN step. Maybe you can keep 'RUN echo' step currently until we find a better way.

Reply:
@liushilongbuaa 
I have this Dockerfile now
```
FROM docker-dpu-base
RUN echo "docker-dpu-base"
```
But still getting the same error.
error squashing image: error getting tar stream to parent: layer ID 'sha256:bb88c4ad01631bb3e6fd8a9ea1721e328bf5fb205f1709663799805a2eff29df' is not a parent of the specified layer: cannot provide diff to non-parent

Reply:
@liushilongbuaa since with this experiment we are getting compilation errors can we go with the committed approach.

Reply:
I'm not very concern on Dockerfile.
Please remove your changes in prepare_docker_buildinfo.sh
Just skip this script at the beginning on SKIP_HOOK tags in Dockerfile.

Reply:
@liushilongbuaa there seems to be some dependency. As i told before if we skip prepare_docker_buildinfo.sh script we are getting error
```
error squashing image: error getting tar stream to parent: layer ID 'sha256:bb88c4ad01631bb3e6fd8a9ea1721e328bf5fb205f1709663799805a2eff29df' is not a parent of the specified layer: cannot provide diff to non-parent
```

Reply:
Undid the changes in prepare_docker_buildinfo.sh

Reply:
I will test this part.
The goal is reduce docker-pdu/Dockerfile.j2 to 3 lines.
prepare_docker_buildinfo.sh add one line to skip pdu docker.

Reply:
Thanks, please let me know the results.